webapp-django/crashstats/profile/templates/profile/profile.html

```diff
@@ -0,0 +1,198 @@
+                    from you in the last 4 weeks</strong>. It's either good news,
+                    or you do not provide your email address when you send us
+                    crash reports.
+                </p>
+
+                {% endif %}
+            </div>
+        </div>
+    </div><!--
+
+ --><div class="page-section">
+        <div class="panel">
+            <div class="title">
+                <h2>API Tokens</h2>
+            </div>
+
+            <div class="body">
+                {% if your_tokens %}
+                <p>Here is a list of your currently generated tokens (<a href="{{ url('tokens:home') }}">manage</a>):</p>
+                {% endif %}
+                {% for token in your_tokens %}
+                <div class="token">
+                    <h4>Generated <time class="ago" data-date="{{ token.created.isoformat() }}">{{ token.created.isoformat() }}</time></h4>
+                    <p class="code">
+                        <code>{{ token.key[:12] }}<span class="rest-hidden">{{ token.key[12:] }}</span><span class="rest-cover">...</span></code>
+                        <button type="button" data-toggle="Hide again">Show the whole token</button>
+                    </p>
+
+                    {% if token.is_expired %}
+                    <p class="is-expired">Token has expired and will not work any more.</p>
+                    {% endif %}
+
+                    <table class="meta-data">
+                        <tr>
+                            <th>Permissions:</th>
+                            <td>
+                                <ul>
+                                    {% for permission in token.permissions.all() %}
+                                        <li><code>{{ permission.name }}</code></li>
+                                    {% else %}
+                                        <li><em>None</em></li>
+                                    {% endfor %}
+                                </ul>
+                            </td>
+                        </tr>
+                        {% if token.is_expired %}
+                        <tr>
+                            <th>Expired:</th>
+                            <td>
+                                {{ token.expires.strftime('%Y-%m-%d %H:%M:%S%Z') }}
+                                (<time class="ago" data-date="{{ token.expires.isoformat() }}">{{ token.expires.isoformat() }}</time>)
+                            </td>
+                        </tr>
+                        {% else %}
+                        <tr>
+                            <th>Expires:</th>
+                            <td>
+                                {{ token.expires.strftime('%Y-%m-%d %H:%M:%S%Z') }}
+                                (<time class="in" data-date="{{ token.expires.isoformat() }}">{{ token.expires.isoformat() }}</time> from now)
+                            </td>
+                        </tr>
+                        {% endif %}
```

Comment:
This is the only change I didn't make, since lines 164 and 172 would have to be if-else as well (the different classes make the text around the dates different).